aea/agent.py

```diff
@@ -28,14 +26,52 @@
 
 from aea.connections.base import Connection
 from aea.identity.base import Identity
+from aea.mail.base import Envelope
 from aea.multiplexer import InBox, OutBox
 from aea.runtime import AsyncRuntime, BaseRuntime, RuntimeStates, ThreadedRuntime
 
 
 logger = logging.getLogger(__name__)
 
 
-class Agent(ABC):
+class AbstractAgent(ABC):
+    """This class provides an abstract base  interface for an agent!"""
+
+    @abstractmethod
+    def setup(self) -> None:
+        """
+        Set up the agent.
+
+        :return: None
+        """
+
+    @abstractmethod
+    def act(self) -> None:
+        """
+        Perform actions on period.
+
+        :return: None
+        """
+
+    @abstractmethod
```

Comment:
Just optically; teardown should be below handle_envelope

Reply:
done